### PR TITLE
Update Keychron boards Part 2

### PR DESCRIPTION
**Pull request by adophoxia**
## Description

This is Part 2 of a 5-part series of PRs to updating the entire Keychron lineup to current QMK standards. Since doing 1 PR to update 25 keyboards is unmanageable, doing them in batches would help make it tolerable. (What is changed can be shown in Part 1 below.)

* #21406 
### Keyboard-specific
- Q6
  - Since the non-encoder variants use 2 extra pins (A2 and C14) for the columns vs. not having them in the encoder variants, the `matrix_pins` must be specified in each variant's `info.json` seperately.
## Part 1 involves updating the following KC boards:
- [x] Q5
- [x] Q6
- [x] Q7
- [x] Q8
- [x] Q9 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/keychron/q5/info.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "rgblight": false
+    },
+    "matrix_pins": {
+        "cols": ["A10", "A9", "A8", "B1", "B0", "A7", "A6", "A5", "A4", "A3", null, null, null, null, null, null, null, null],
+        "custom": true,
+        "custom_lite": true,
+        "rows": ["B5", "B4", "B3", "A15", "A14", "A13"]
```
Comment (fauxpark) [inline keyboards/keychron/q5/info.json:24]: Please leave custom amtrix config as is - col and row pins for custom matrix should not be in info.json as one or both may not actually exist.

File: keyboards/keychron/q5/info.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    },
+    "matrix_size": {
+        "cols": 18,
+        "rows": 6
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q5/info.json:29]:
```suggestion
```
etc.

File: keyboards/keychron/q5/info.json (hunk 1 above)
Comment (adophoxia) [reply keyboards/keychron/q5/info.json:24]: Seeing how I have this set up for every other board that uses custom matrix (in this PR, the Q6), does that mean having to put them back into `config.h` for custom matrix config?

Comment (adophoxia) [reply keyboards/keychron/q5/info.json]: Taken care of in https://github.com/qmk/qmk_firmware/pull/21423/commits/e87a2c37be06c34fbb0ad3045703014537262d00.

File: keyboards/keychron/q5/info.json (hunk 2 above)
Comment (adophoxia) [reply keyboards/keychron/q5/info.json]: As above.

**Comment by linouxis9** [comment]
Looking for the Q6 to be merged, thanks a lot for your refactoring!!

**Comment by adophoxia** [comment]
Q6 is already in master.

File: keyboards/keychron/q5/config.h [hunk 3]
```diff
@@ -46,58 +43,17 @@
 /* turn off effects when suspended */
 #define RGB_DISABLE_WHEN_USB_SUSPENDED
 
+/* Encoder Configuration*/
+#ifdef ENCODER_ENABLE
+    #define ENCODER_DEFAULT_POS 0x3
+#endif
```
Comment (drashna) [inline keyboards/keychron/q5/config.h:49]:
```suggestion
#define ENCODER_DEFAULT_POS 0x3

```

File: keyboards/keychron/q5/info.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            "solid_splash": true,
+            "splash": true,
+            "typing_heatmap": true
+        },
+        "driver": "CKLED2001"
+    },
+    "url": "https://github.com/Keychron",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0150",
```
Comment (drashna) [inline keyboards/keychron/q5/info.json:50]: If this is specified here, then it doesn't need to be specified at each subfolder.  Or vice versa.

File: keyboards/keychron/q6/config.h [hunk 5]
```diff
@@ -40,58 +37,17 @@
 /* turn off effects when suspended */
 #define RGB_DISABLE_WHEN_USB_SUSPENDED
 
+/* Encoder Configuration */
+#ifdef ENCODER_ENABLE
+    #define ENCODER_DEFAULT_POS 0x3
+#endif
```
Comment (drashna) [inline keyboards/keychron/q6/config.h:43]:
```suggestion
#define ENCODER_DEFAULT_POS 0x3
```

File: keyboards/keychron/q8/config.h [hunk 6]
```diff
@@ -42,62 +42,20 @@
 /* Disable DIP switch in matrix data */
 #define MATRIX_MASKED
 
+/* Encoder Configuration */
+#ifdef ENCODER_ENABLE
+    #define ENCODER_DEFAULT_POS 0x3
+#endif
```
Comment (drashna) [inline keyboards/keychron/q8/config.h:48]:
```suggestion
#define ENCODER_DEFAULT_POS 0x3
```

File: keyboards/keychron/q9/config.h [hunk 7]
```diff
@@ -39,7 +39,10 @@
 /* turn off effects when suspended */
 #define RGB_DISABLE_WHEN_USB_SUSPENDED
 
-#define DYNAMIC_KEYMAP_LAYER_COUNT 5
+/* Encoder Configuration */
+#ifdef ENCODER_ENABLE
+    #define ENCODER_DEFAULT_POS 0x3
+#endif
```
Comment (drashna) [inline keyboards/keychron/q9/config.h:45]:
```suggestion
#define ENCODER_DEFAULT_POS 0x3
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by adophoxia** [comment]
Closing this PR for the sake of making each board have its own PR for sanity.

**Pull request closed without merging**